lib/kino/js/live.ex

```
@@ -0,0 +1,369 @@
+
+      @before_compile Kino.JS.Live
+    end
+  end
+
+  def __before_compile__(env) do
+    unless Module.defines?(env.module, {:__js_info__, 0}) do
+      message = """
+      make sure to include Kino.JS in #{inspect(env.module)} and define the necessary assets.
+
+          use Kino.JS
+          use Kino.JS.Live
+
+      See Kino.JS for more details.
+      """
+
+      IO.warn(message, Macro.Env.stacktrace(env))
+    end
+
+    nil
+  end
+
+  @doc """
+  Instantiates a live JavaScript widget defined by `module`.
+
+  The given `init_arg` is passed to the `init/2` callback when
+  the underlying widget process is started.
+  """
+  @spec new(module(), term()) :: t()
+  def new(module, init_arg) do
+    {:ok, pid} = Kino.start_child({__MODULE__, {module, init_arg}})
+    %__MODULE__{module: module, pid: pid}
+  end
+
+  @doc false
+  def start_link(opts) do
+    GenServer.start_link(__MODULE__, opts)
+  end
+
+  @doc """
+  Sends an asynchronous request to the widget server.
+
+  See `GenServer.cast/2` for more details.
+  """
+  @spec cast(t(), term()) :: :ok
+  def cast(widget, term) do
+    GenServer.cast(widget.pid, term)
+  end
+
+  @doc """
+  Makes a synchronous call to the widget server and waits
+  for its reply.
+
+  See `GenServer.call/3` for more details.
+  """
+  def call(widget, term, timeout \\ 5_000) do
+    GenServer.call(widget.pid, term, timeout)
+  end
+
+  @impl true
+  def init({module, init_arg}) do
+    ctx = Context.new()
+
+    {:ok, ctx} =
+      if has_function?(module, :init, 2) do
+        module.init(init_arg, ctx)
+      else
+        {:ok, ctx}
+      end
+
+    {:ok, %{module: module, client_pids: [], client_monitor_refs: [], ctx: ctx}}
+  end
+
+  @impl true
+  def handle_cast(msg, state) do
+    unless has_function?(state.module, :handle_cast, 2) do
+      raise "attempted to cast #{inspect(__MODULE__)}, but no handle_cast/2 was defined #{inspect(state.module)}"
+    end
```

You don't need this, you can just call it and let it crash!